### ISSUE TITLE
Support for @sap/cds 5.9.0

### ISSUE THREAD

**Issue by gregorwolf** [sapmentors/cds-pg#288]
Today the @sap/cds version 5.9.0 was published. Unfortunately when updating cds-pg to this version already the simple http://localhost:4004/beershop/Beers request fails with this error:

```
[INTERNAL ERROR] TypeError: Cannot set property 'undefined' of undefined
    at /Users/gwolf/Documents/Projects/cap/cds-pg/node_modules/@sap/cds/libx/_runtime/db/data-conversion/post-processing.js:162:36
    at Array.<anonymous> (/Users/gwolf/Documents/Projects/cap/cds-pg/lib/pg/data-conversion/post-processing.js:80:23)
    at _processRow (/Users/gwolf/Documents/Projects/cap/cds-pg/lib/pg/data-conversion/post-processing.js:158:14)
    at postProcess (/Users/gwolf/Documents/Projects/cap/cds-pg/lib/pg/data-conversion/post-processing.js:181:7)
    at _executeSQLReturningRows (/Users/gwolf/Documents/Projects/cap/cds-pg/lib/pg/execute.js:227:10)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async next (/Users/gwolf/Documents/Projects/cap/cds-pg/node_modules/@sap/cds/lib/serve/Service-dispatch.js:74:17)
    at async PostgresDatabase.handle (/Users/gwolf/Documents/Projects/cap/cds-pg/node_modules/@sap/cds/lib/serve/Service-dispatch.js:72:10)
    at async next (/Users/gwolf/Documents/Projects/cap/cds-pg/node_modules/@sap/cds/lib/serve/Service-dispatch.js:74:17)
    at async ApplicationService.handle (/Users/gwolf/Documents/Projects/cap/cds-pg/node_modules/@sap/cds/lib/serve/Service-dispatch.js:72:10)
```

hope @sjvans has a tip for us.

**Comment by sebastianesch** [comment]
It seems `getPostProcessMapper` has changed from @sap/cds 5.8.4 to 5.9 in a way that breaks cds-pg:

5.8.4:

```
const getPostProcessMapper = (toService, csn = {}, cqn = {}) => {
  // No mapper defined or irrelevant as no READ request
  if (!Object.prototype.hasOwnProperty.call(cqn, 'SELECT')) {
    return new Map()
  }

  if (Array.isArray(cqn.SELECT.columns) && cqn.SELECT.columns.length !== 0 && !cqn.SELECT.columns.includes('*')) {
    return _getMapperForListedElements(toService, csn, cqn)
  }

  // No element/column specified
  return _getElementCombinations(toService, csn, cqn)
}
```

5.9
```
const getPostProcessMapper = (conversionMap, csn = {}, cqn = {}) => {
  return cqn.SELECT.columns ? _getMapperForListedElements(conversionMap, csn, cqn) : new Map()
}
```

The 5.9 version only works for SELECT queries, but cds-pg uses it generic queries, as well as for select and insert queries.

Kind regards,
Sebastian

**Comment by rbhuva** [comment]
Hi @gregorwolf,

Should we remove carat sign in dependency of @sap/cds because it is picking up 5.9 while npm install thus breaking cds-pg ?

Thanks,
Rajdeep Bhuva

**Comment by sjvans** [comment]
hi @gregorwolf 

code went through a cleanup/ refactoring. `lib/pg/data-conversion/post-processing.js` looks like a copy of cds code, so you might simply be able to update the copy to make it work again. if not i can take a look, but it might be a while since i'm on parental leave at the moment...

best,
sebastian

**Comment by johannes-vogel** [comment]
@gregorwolf We have released version 5.9.1 which might fix this issue. If not, we'd need to have a closer look.
